src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/MessagePack/Formatters/RazorProjectInfoFormatter.cs

```diff
@@ -30,7 +30,7 @@ public override RazorProjectInfo Deserialize(ref MessagePackReader reader, Seria
         var serializedFilePath = CachedStringFormatter.Instance.Deserialize(ref reader, options).AssumeNotNull();
         var filePath = CachedStringFormatter.Instance.Deserialize(ref reader, options).AssumeNotNull();
         var configuration = reader.DeserializeOrNull<RazorConfiguration>(options);
-        var projectWorkspaceState = reader.DeserializeOrNull<ProjectWorkspaceState>(options);
+        var projectWorkspaceState = reader.Deserialize<ProjectWorkspaceState>(options);
```

Comment:
Shouldn't this be `DeserializeOrNull<ProjectWorkspaceState>(options) ?? ProjectWorkspaceState.Default`, otherwise we need to bump the version number?

Same for configuration (and possibly json?)

Reply:
Great catch! I'll update this as you suggested. I'll also create an issue to remove these tweaks the next time that we are bumping the version number.